src/model/Timestep.cpp

```diff
@@ -66,11 +68,20 @@ namespace model {
     }
 
     bool Timestep_Impl::setNumberOfTimestepsPerHour(int numberOfTimestepsPerHour) {
+      int currentNumberOfTimestepsPerHour = numberOfTimestepsPerHour();
+      if (currentNumberOfTimestepsPerHour != numberOfTimestepsPerHour) {
+        for (auto& sch_day : model().getConcreteModelObjects<ScheduleDay>()) {
+          sch_day.getImpl<ScheduleDay_Impl>()->clearCachedVariables();
+        }
```

Comment:
I don't think you need to clear the cached times and values. Just the timeseries